kernel/thread.c

```diff
@@ -813,3 +827,23 @@ size_t myst_kill_thread_group()
 
     return count;
 }
+
+int myst_set_thread_name(myst_thread_t* thread, const char* n)
+{
+    int ret = 0;
+
+    if (!thread || !n)
+        ERAISE(-EINVAL);
+
+    /* Copy atmost 15 bytes */
+    strncpy(thread->name, n, sizeof(thread->name) - 1);
+    /* Set null terminator if string pointed by n
+        longer than the 16-byte fixed buffer for thread->name */
```

Comment:
Is this the same behavior as Linux?

Reply:
Yes. According to the man prctl page - the name can be atmost 16 bytes, including the null termination character.